### ISSUE TITLE
GBM doesn't like ":" in column name

### ISSUE THREAD

**Issue by jmrichardson** [awslabs/autogluon#1033]
Using the [tutorial](https://auto.gluon.ai/stable/tutorials/tabular_prediction/tabular-indepth.html) with a different data set produced the error below.  Changing the column names to exclude a ":" resolved the issue however would like to report it as I prefer them and the other models i have tested so far don't seem to have this issue.  Also, this only occurs with something similar to the the following in fit (ie commenting them out does not produce the error).

```
    num_bag_folds=5,
    num_bag_sets=2,  
    num_stack_levels=2,
```

This resolved the problem:

`train_data.columns = [re.sub(r':', '_', col) for col in train_data.columns]`

Here is the error:

```
Fitting model: LightGBM_BAG_L1\T0 ... Training model for up to 30.39s of the 96.97s of remaining time.
	Warning: Exception caused LightGBM_BAG_L1\T0 to fail during training... Skipping this model.
		"None of [Index(['close', 'pta.hilo_high_length:1019_low_length:6646_0',\n       'pta.hilo_high_length:1019_low_length:6646_1',\n       'pta.hilo_high_length:1019_low_length:6646_2',\n       'tta.MIDPOINT_timeperiod:5952', 'pta.pwma_length:6042',\n       'pta.pvi_length:5903_initial:1241', 'fta.SMM_period:6867',\n       'fta.VAMA_period:6760', 'fta.EVWMA_period:4723',\n       'tta.T3_timeperiod:4868', 'pta.ttm_trend_length:6119',\n       'pta.vidya_length:416', 'tta.BBANDS_timeperiod:6238_0',\n       'tta.BBANDS_timeperiod:6238_1', 'tta.BBANDS_timeperiod:6238_2',\n       'fta.BBANDS_period:4424_0', 'fta.BBANDS_period:4424_1',\n       'fta.BBANDS_period:4424_2',\n       'fta.KAMA_er:4985_ema_fast:1498_ema_slow:152_period:7420',\n       'pta.decreasing_length:3031', 'fta.EFI_period:1761'],\n      dtype='object')] are in the [columns]"
Detailed Traceback:
Traceback (most recent call last):
  File "D:\Projects\tactory\autogluon\tabular\trainer\abstract_trainer.py", line 911, in _train_and_save
    model = self._train_single(X, y, model, X_val, y_val, **model_fit_kwargs)
  File "D:\Projects\tactory\autogluon\tabular\trainer\abstract_trainer.py", line 881, in _train_single
    model.fit(X=X, y=y, **model_fit_kwargs)
  File "D:\Projects\tactory\autogluon\core\models\abstract\abstract_model.py", line 405, in fit
    self._fit(**kwargs)
  File "D:\Projects\tactory\autogluon\core\models\ensemble\stacker_ensemble_model.py", line 142, in _fit
    super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)
  File "D:\Projects\tactory\autogluon\core\models\ensemble\bagged_ensemble_model.py", line 216, in _fit
    fold_model.fit(X=X_fold, y=y_fold, X_val=X_val_fold, y_val=y_val_fold, time_limit=time_limit_fold, **kwargs_fold)
  File "D:\Projects\tactory\autogluon\core\models\abstract\abstract_model.py", line 405, in fit
    self._fit(**kwargs)
  File "D:\Projects\tactory\autogluon\tabular\models\lgb\lgb_model.py", line 92, in _fit
    sample_weight=sample_weight, sample_weight_val=sample_weight_val, dataset_train=dataset_train, dataset_val=dataset_val)
  File "D:\Projects\tactory\autogluon\tabular\models\lgb\lgb_model.py", line 257, in generate_datasets
    X = self.preprocess(X, is_train=True)
  File "D:\Projects\tactory\autogluon\core\models\abstract\abstract_model.py", line 250, in preprocess
    X = self._preprocess_nonadaptive(X, **kwargs)
  File "D:\Projects\tactory\autogluon\tabular\models\lgb\lgb_model.py", line 234, in _preprocess_nonadaptive
    X = super()._preprocess_nonadaptive(X=X, **kwargs)
  File "D:\Projects\tactory\autogluon\core\models\abstract\abstract_model.py", line 284, in _preprocess_nonadaptive
    X = X[self.features]
  File "C:\Users\john\AppData\Roaming\Python\Python37\site-packages\pandas\core\frame.py", line 3030, in __getitem__
    indexer = self.loc._get_listlike_indexer(key, axis=1, raise_missing=True)[1]
  File "C:\Users\john\AppData\Roaming\Python\Python37\site-packages\pandas\core\indexing.py", line 1266, in _get_listlike_indexer
    self._validate_read_indexer(keyarr, indexer, axis, raise_missing=raise_missing)
  File "C:\Users\john\AppData\Roaming\Python\Python37\site-packages\pandas\core\indexing.py", line 1308, in _validate_read_indexer
    raise KeyError(f"None of [{key}] are in the [{axis_name}]")
KeyError: "None of [Index(['close', 'pta.hilo_high_length:1019_low_length:6646_0',\n       'pta.hilo_high_length:1019_low_length:6646_1',\n       'pta.hilo_high_length:1019_low_length:6646_2',\n       'tta.MIDPOINT_timeperiod:5952', 'pta.pwma_length:6042',\n       'pta.pvi_length:5903_initial:1241', 'fta.SMM_period:6867',\n       'fta.VAMA_period:6760', 'fta.EVWMA_period:4723',\n       'tta.T3_timeperiod:4868', 'pta.ttm_trend_length:6119',\n       'pta.vidya_length:416', 'tta.BBANDS_timeperiod:6238_0',\n       'tta.BBANDS_timeperiod:6238_1', 'tta.BBANDS_timeperiod:6238_2',\n       'fta.BBANDS_period:4424_0', 'fta.BBANDS_period:4424_1',\n       'fta.BBANDS_period:4424_2',\n       'fta.KAMA_er:4985_ema_fast:1498_ema_slow:152_period:7420',\n       'pta.decreasing_length:3031', 'fta.EFI_period:1761'],\n      dtype='object')] are in the [columns]"
Fitting model: LightGBM_BAG_L1\T1 ... Training model for up to 30.37s of the 96.95s of remaining time.
	Warning: Exception caused LightGBM_BAG_L1\T1 to fail during training... Skipping this model.
		"None of [Index(['close', 'pta.hilo_high_length:1019_low_length:6646_0',\n       'pta.hilo_high_length:1019_low_length:6646_1',\n       'pta.hilo_high_length:1019_low_length:6646_2',\n       'tta.MIDPOINT_timeperiod:5952', 'pta.pwma_length:6042',\n       'pta.pvi_length:5903_initial:1241', 'fta.SMM_period:6867',\n       'fta.VAMA_period:6760', 'fta.EVWMA_period:4723',\n       'tta.T3_timeperiod:4868', 'pta.ttm_trend_length:6119',\n       'pta.vidya_length:416', 'tta.BBANDS_timeperiod:6238_0',\n       'tta.BBANDS_timeperiod:6238_1', 'tta.BBANDS_timeperiod:6238_2',\n       'fta.BBANDS_period:4424_0', 'fta.BBANDS_period:4424_1',\n       'fta.BBANDS_period:4424_2',\n       'fta.KAMA_er:4985_ema_fast:1498_ema_slow:152_period:7420',\n       'pta.decreasing_length:3031', 'fta.EFI_period:1761'],\n      dtype='object')] are in the [columns]"
```

**Comment by Innixma** [comment]
Very nice catch. This is an extremely specific error that is due to how models are trained when bagging, which bypasses the correct conversion of feature column names in LightGBM when invalid column names are present. For now, I'd recommend cleaning the model names outside of AutoGluon before sending them in, and I will look into fixing this.

**Comment by Innixma** [comment]
This is now fixed in #1036. Thanks again for reporting!
